### PR TITLE
fix: initial yield path simplification test updates

### PR DESCRIPTION
**Pull request by axejintao**
# Summary

- removes unused paths (and updates to paths) from yield maths
- fixes test leaks

### CODE REVIEW

File: src/indexers/vault-definition-indexer.spec.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ describe('vault-definition-indexer', () => {
         return { sett: <gqlGenT.SettQuery['sett']>graphVault, __typename: 'Query' };
       });
 
-      await mockSupportedChains();
```
Comment (NamelsKing) [inline src/indexers/vault-definition-indexer.spec.ts:38]: why u have removed this one? looks like `setupMockChain` mocks only 1 chain, and that helper was mocking `sdk`  and providers for all available chains

File: src/vaults/yields.utils.ts [hunk 2]
```diff
@@ -463,3 +432,153 @@ function constructGraphVaultData(
     };
   });
 }
+
+export async function estimateVaultPerformance(
```
Comment (axejintao) [inline src/vaults/yields.utils.ts:436]: this is the main non-test related (re enable) change for this pr

File: src/vaults/yields.utils.ts [hunk 3]
```diff
@@ -463,3 +432,153 @@ function constructGraphVaultData(
...
+  chain: Chain,
+  vault: VaultDefinitionModel,
+  data: VaultHarvestData[],
+): Promise<YieldSource[]> {
+  if (data.length === 0) {
+    return [];
+  }
+
+  const sdk = await chain.getSdk();
+  const recentHarvests = data.sort((a, b) => b.timestamp - a.timestamp);
```
Comment (axejintao) [inline src/vaults/yields.utils.ts:446]: new behavior of the yield function is to always assume the period is 15 days - this makes it simpler for the function (should capture in a comment above) to just base all the yield within one window

File: src/vaults/yields.utils.ts [hunk 4]
```diff
@@ -463,3 +432,153 @@ function constructGraphVaultData(
...
+
+  let harvestReport = `${vault.name} Harvest Report\n`;
+
+  let totalHarvested = 0;
+  let totalAccumulated = 0;
+  let totalVaultTokens = 0;
+  let totalVaultPrincipal = 0;
+
+  console.log(`Evaluating ${allEvents.length} events for ${vault.name}`);
+  for (const event of allEvents) {
```
Comment (axejintao) [inline src/vaults/yields.utils.ts:490]: parse through all events to capture and average capital yielding the given captured rewards,
here is where we would persist this input data to display on a 'yield inputs' page

File: src/vaults/yields.utils.ts [hunk 5]
```diff
@@ -463,3 +432,153 @@ function constructGraphVaultData(
...
+    totalVaultPrincipal += vaultPrice * balance;
+
+    harvestReport = harvestReport.concat(
+      `[${new Date(event.timestamp * 1000).toDateString()}] ${amount.toFixed(2)} ${
+        token.symbol
+      } ($${tokenEarned.toFixed(2)}) - Balance: ${balance.toFixed(2)}\n`,
+    );
+  }
+
+  const duration = VAULT_TWAY_PERIOD * ONE_DAY_MS;
```
Comment (axejintao) [inline src/vaults/yields.utils.ts:520]: here, the duration is always calculated as the set period

**Review by NamelsKing — APPROVED** [review]

File: src/vaults/yields.utils.ts [hunk 6]
```diff
@@ -463,3 +432,153 @@ function constructGraphVaultData(
...
+  const averagePrincipal = totalVaultPrincipal / allEvents.length;
+  const averageTokens = totalVaultTokens / allEvents.length;
+  const apr = calculateYield(averagePrincipal, totalAccumulated, duration);
```
Comment (axejintao) [inline src/vaults/yields.utils.ts:523]: we calculate (for reporting purposes) the evaluated apr using the yield functions

File: src/vaults/yields.utils.ts [hunk 7]
```diff
@@ -463,3 +432,153 @@ function constructGraphVaultData(
...
+  const valueSources = [];
+
+  // create the apr source for harvests
+  const compoundApr = calculateYield(totalVaultTokens, totalHarvested, duration);
+  const compoundYieldSource = createYieldSource(vault, SourceType.PreCompound, VAULT_SOURCE, compoundApr);
+  valueSources.push(compoundYieldSource);
+
+  // create the apy source for harvests
+  const compoundApy = calculateYield(totalVaultTokens, totalHarvested, duration, totalHarvested);
+  const compoundedYieldSource = createYieldSource(vault, SourceType.Compound, VAULT_SOURCE, compoundApy);
```
Comment (axejintao) [inline src/vaults/yields.utils.ts:540]: updated all code to utilize new same yield calculation functions

File: src/vaults/yields.utils.ts [hunk 8]
```diff
@@ -463,3 +432,153 @@ function constructGraphVaultData(
...
+    if (!entry) {
+      entry = BigNumber.from(0);
+      tokensEmitted.set(token, entry);
+    }
+    tokensEmitted.set(token, entry.add(amount));
+  }
+
+  let flywheelCompounding = 0;
+
+  for (const [token, amount] of tokensEmitted.entries()) {
```
Comment (axejintao) [inline src/vaults/yields.utils.ts:556]: this is old code, will suffice for now - likely need to do another round to utilize the captures above vs. this as we are losing time granularity here (and thus subject to price fluctuation inaccuracies)

**Pull request merged**